dvc/remote/base.py

```diff
@@ -70,11 +72,24 @@ def __init__(self, checksum):
         )
 
 
+def index_locked(f):
```

Comment:
Do we really need this lock, btw? index is per-repo, so as long as repo is locked - index is locked too. And I don't see us unlocking `.dvc/lock` for status/gc/pull/pull (at least right now).

Reply:
The sqlite3 python lib isn't thread safe, or at least it's not safe to share cursor objects across threads, so it's more to make sure the index is only being accessed from the main thread

Reply:
In terms of what's in this PR the lock isn't needed, but since there are parts of `cache_exists` which are multithreaded, it seems safer to prevent anyone from accidentally misusing the `remote.index` object if/when `status` or `cache_exists` behavior is updated in the future

Reply:
@pmrowla Got it, makes sense to me. Thanks!